### ISSUE TITLE
Wrong type inference for class with metaclass that acts as descriptor

### ISSUE THREAD

**Issue by jack1142** [microsoft/pyright#2164]
**Describe the bug**
Descriptor protocol does not work on metaclasses - defining `__get__` method in a metaclass does not make classes using that metaclass work as descriptors.

**To Reproduce**
1. Copy code from below
2. Run it, you should see:
```
123
123
123
123
```
3. Type check it with pyright, as part of the output you should see:
```
metaclass_descriptors.py:28:17 - info: Type of "X.number_cls" is "Type[IntDescriptorClass]"
metaclass_descriptors.py:29:17 - info: Type of "X().number_cls" is "Type[IntDescriptorClass]"
metaclass_descriptors.py:30:17 - info: Type of "X.number" is "int"
metaclass_descriptors.py:31:17 - info: Type of "X().number" is "int"
```

**Expected behavior**
I expected pyright to infer type of `X.number_cls`/`X().number_cls` correctly:
```
metaclass_descriptors.py:28:17 - info: Type of "X.number_cls" is "int"
metaclass_descriptors.py:29:17 - info: Type of "X().number_cls" is "int"
metaclass_descriptors.py:30:17 - info: Type of "X.number" is "int"
metaclass_descriptors.py:31:17 - info: Type of "X().number" is "int"
```
Note:
To simplify this example, `__get__` returns an `int` no matter if `instance` is `None` or not but I assume a fix would also make it work properly with overloads as it does for instances of classes.

**Screenshots or Code**
```python
from typing import Any, TYPE_CHECKING, overload

class _IntDescriptorMeta(type):
    def __get__(self, instance: Any, owner: Any) -> int:
        return 123

class IntDescriptorClass(metaclass=_IntDescriptorMeta):
    ...

class IntDescriptor:
    def __get__(self, instance: Any, owner: Any) -> int:
        return 123

class X:
    number_cls = IntDescriptorClass
    number = IntDescriptor()

print(X.number_cls)
print(X().number_cls)
print(X.number)
print(X().number)

if TYPE_CHECKING:
    reveal_type(X.number_cls)
    reveal_type(X().number_cls)
    reveal_type(X.number)
    reveal_type(X().number)
```

**VS Code extension or command-line**
Running pyright 1.1.160 as a command-line tool

**Additional context**
N/A

**Comment by erictraut** [comment]
Interesting. The current behavior in pyright is consistent with mypy, but it looks like neither of them match the runtime behavior in the Python interpreter.

**Comment by erictraut** [comment]
I've added support for metaclasses that provide descriptor methods. This will be in the next release of pyright.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.161, which I just published. It will also be included in the next release of pylance.
